### ISSUE TITLE
Mocked `fungibles::*` traits

### ISSUE THREAD

**Issue by ggwpez** [paritytech/substrate#14534]
Instead of always using `pallet-balances` in mock runtimes we should rather write a mock for the `fungibles::*` traits.

**Comment by muraca** [comment]
I can start working on this if you want.
Is there any example of mock being implemented for other traits?

**Comment by ggwpez** [comment]
Yes @muraca, although I currently dont have much time to review things.  

One example would be the account storage map [here](https://github.com/paritytech/substrate/blob/79d37ef461a4bdcf6ede7b37d64b28b58484dc7c/frame/balances/src/tests/mod.rs#L235), or the [Polling](https://github.com/paritytech/substrate/blob/79d37ef461a4bdcf6ede7b37d64b28b58484dc7c/frame/conviction-voting/src/tests.rs#L117). They all use thread locals via `parameter_types!` and `pub static`.  
I think this should also be used here. Thinking about this, you will probably have to build some raw byte storage with thread locals and then decode the correct type out of them, since thread locals cannot be generic but the wrapper that accesses them is generic...

**Comment by ggwpez** [comment]
Actually @gavofyork is working on it himself (draft: https://gist.github.com/gavofyork/0e547f7adf26f69b2753e3cfa9ebcf55). So maybe wait there is a draft MR and then extend the missing traits or see what is needed afterwards.
